From ceb9783179c1002acb61bfb2d5a1801f450c2b37 Mon Sep 17 00:00:00 2001
From: Bryce Harrington <bryce@osg.samsung.com>
Date: Wed, 20 Sep 2017 16:05:54 -0700
Subject: [PATCH cairo] polygon-intersection: Check for invalid right edge

Signed-off-by: Bryce Harrington <bryce@osg.samsung.com>
---
 src/cairo-polygon-intersect.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cairo-polygon-intersect.c b/src/cairo-polygon-intersect.c
index 9c1777f..6fab924 100644
--- a/src/cairo-polygon-intersect.c
+++ b/src/cairo-polygon-intersect.c
@@ -1166,6 +1166,8 @@ active_edges (cairo_bo_edge_t		*left,
 	    } while (1);
 
 	    right = left->next;
+	    if (! right)
+		return;
 	    do {
 		if unlikely ((right->deferred.other))
 		    edges_end (right, top, polygon);
-- 
2.7.4

